ouroboros-consensus/src/Ouroboros/Consensus/NodeNetwork.hs

```diff
@@ -277,55 +274,74 @@ protocolCodecs cfg version = ProtocolCodecs {
     , pcLocalStateQueryCodec      = staticLocalStateQueryCodec
     }
 
+data NodeToClientCodecs blk failure m bytesLCS bytesLTX bytesLSQ = NodeToClientCodecs {
+    nodeToClientChainSyncCodec       :: Codec (ChainSync (Serialised blk) (Tip blk))
+                                         failure m bytesLCS
+  , nodeToClientTxSubmissionCodec    :: Codec (LocalTxSubmission (GenTx blk) (ApplyTxErr blk))
+                                         failure m bytesLTX
+  , nodeToClientStateQueryCodec      :: Codec (LocalStateQuery blk (Query blk))
+                                         failure m bytesLSQ
+  }
+
+nodeToClientCodecs
+    :: forall m blk. (IOLike m, RunNode blk)
+    => NodeToClientVersion
+    -> NodeToClientCodecs blk DeserialiseFailure m ByteString ByteString ByteString
+nodeToClientCodecs NodeToClientV_1 = NodeToClientCodecs {
+    nodeToClientChainSyncCodec    = staticLocalChainSyncCodec
+  , nodeToClientTxSubmissionCodec = staticLocalTxSubmissionCodec
+  , nodeToClientStateQueryCodec   = staticLocalStateQueryCodec
+  }
+
 staticBlockFetchCodecSerialised
```

Comment:
Please hide them in a `where` clause`, then you can rename them as `nodeToClientCHainSyncCodec`, ... and take advantage of `NamedFieldPuns` extension:
```
nodeToClientCodecs NodeToClientV_1 = NodeToClientCodecs {
      nodeToClientChainSyncCodec
    , nodeToClientTxSubmissionCodec
    , nodeToClientStateQueryCodec
    }
  where
    nodeToClientChainSyncCodec = ...
    nodeToClientTxSubmissionCodec = ...
    nodeToClientStateQueryCodec = ...
```

Reply:
At the moment the same codecs are exported via two different ways via `ProtocolCodecs` and via `NodeToClientCodecs`.
Maybe the nodeToClient codecs should be removed from ProtocolCodecs altogther ?
`staticLocalChainSyncCodec` is the single definition that is uses for both exports.

Reply:
I see; you should have access to `NodeToClientV_1` there too, no? Then you can unpack the record and add the codes.  This way in the future one will only need to modify `nodeToClientCodecs` to update codecs in consensus.

Reply:
OK! Done